### PR TITLE
feat(pair_astro_xastro, bonded template): Add pair ASTRO-xASTRO and bonded template

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

File: templates/pair_bonded_template/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+crate-type = ["cdylib", "rlib"]
+
+[features]
+# for quicker tests, cargo test --lib
+# for more explicit tests, cargo test --features=backtraces
+backtraces = ["cosmwasm-std/backtraces"]
+library = []
+
+[dependencies]
+integer-sqrt = "0.1.5"
```
Comment (epanchee) [inline templates/pair_bonded_template/Cargo.toml:27]: This import is unused

File: contracts/pair_astro_xastro/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+crate-type = ["cdylib", "rlib"]
+
+[features]
+# for quicker tests, cargo test --lib
+# for more explicit tests, cargo test --features=backtraces
+backtraces = ["cosmwasm-std/backtraces"]
+library = []
+
+[dependencies]
+integer-sqrt = "0.1.5"
```
Comment (epanchee) [inline contracts/pair_astro_xastro/Cargo.toml:27]: This import is unused

File: contracts/pair_astro_xastro/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            messages.push(CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: params.astro_addr.to_string(),
+                msg: to_binary(&Cw20ExecuteMsg::Send {
+                    contract: params.staking_addr.to_string(),
+                    amount: offer_asset.amount,
+                    msg: to_binary(&StakingCw20HookMsg::Enter {})?,
+                })?,
+                funds: vec![],
+            }))
+        } else if offer_asset.info.equal(&AssetInfo::Token {
```
Comment (epanchee) [inline contracts/pair_astro_xastro/src/contract.rs:90]: This `if` is redundant as you already check that both assets belong to the pair in lines 64-72. I think you may just leave `else ...` here

File: contracts/pair_astro_xastro/src/contract.rs [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        deps: Deps,
+        _env: Env,
+        offer_asset: Asset,
+    ) -> StdResult<SimulationResponse> {
+        let config: Config = CONFIG.load(deps.storage)?;
+        let pools: [AssetInfo; 2] = config.pair_info.asset_infos;
+
+        if !offer_asset.info.equal(&pools[0]) && !offer_asset.info.equal(&pools[1]) {
+            return Err(StdError::generic_err(
+                "Given offer asset doesn't belong to pairs",
```
Comment (epanchee) [inline contracts/pair_astro_xastro/src/contract.rs:139]:
```suggestion
                "Given offer asset doesn't belong to pair",
```

File: templates/pair_bonded_template/Cargo.toml (hunk 1 above)
Comment (maksk7) [reply templates/pair_bonded_template/Cargo.toml]: Removed

File: contracts/pair_astro_xastro/Cargo.toml (hunk 2 above)
Comment (maksk7) [reply contracts/pair_astro_xastro/Cargo.toml]: Removed

File: contracts/pair_astro_xastro/src/contract.rs (hunk 3 above)
Comment (maksk7) [reply contracts/pair_astro_xastro/src/contract.rs]: Ok, changed

File: contracts/pair_astro_xastro/src/contract.rs (hunk 4 above)
Comment (maksk7) [reply contracts/pair_astro_xastro/src/contract.rs]: Changed, thanks

**Review by ShadoySV — CHANGES_REQUESTED** [review]

File: contracts/pair_astro_xastro/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+[features]
+# for quicker tests, cargo test --lib
+# for more explicit tests, cargo test --features=backtraces
+backtraces = ["cosmwasm-std/backtraces"]
+library = []
+
+[dependencies]
+astroport = { path = "../../packages/astroport", default-features = false }
+astroport-pair-bonded-template = { path = "../../templates/pair_bonded_template" }
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/pair_astro_xastro/Cargo.toml]: Done, now pair-bonded is a separated package and created template

File: templates/pair_bonded_template/src/base.rs [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+    /// ## Params
+    /// * **belief_price** is an object of type [`Option<Decimal>`]. This is the belief price used in the swap.
+    ///
+    /// * **max_spread** is an object of type [`Option<Decimal>`]. This is the
+    /// max spread allowed so that the swap can be executed successfuly.
+    ///
+    /// * **offer_amount** is an object of type [`Uint128`]. This is the amount of assets to swap.
+    ///
+    /// * **return_amount** is an object of type [`Uint128`]. This is the amount of assets to receive from the swap.
+    fn assert_max_spread(
```
[the remark this answers is not in the document]
Comment (maksk7) [reply templates/pair_bonded_template/src/base.rs]: Done, thanks

**Review by ShadoySV — APPROVED** [review]

**Review by P-Yevhenii — APPROVED** [review]

**Review by epanchee — APPROVED** [review]

**Pull request merged**